pkg/controller/jenkins/controller.go

```diff
@@ -10,6 +10,7 @@ import (
 	imagev1 "github.com/openshift/api/image/v1"
 	routev1 "github.com/openshift/api/route/v1"
 	jenkinsv1alpha1 "github.com/redhat-developer/openshift-jenkins-operator/pkg/apis/jenkins/v1alpha1"
+	utils "github.com/redhat-developer/openshift-jenkins-operator/pkg/utils"
```

Comment:
We usually try to avoid packages like `utils`, `commons` since they end up becoming a sink-hole of unrelated things. If you need to have `Error` types defined, create a package called `errors` and put everything in there.

Reply:
In case of Reconciler.Errors I feel the need to have a Errors struct more personal to the reconciler instead of utils definitely.

Reply:
I renamed it to `package errors` per @sbose78 suggestion. I also didn't like the utils package.

@waveywaves The idea would be to have a multi-errors wrapper in case of "non-lethal operations" , we "chain" them, and at the end we check if errors.hasErrors() and then process it or log multiple error messages instead of one.

So, if we could reuse it outside of the scope of reconcilers it is good also.

Reply:
I assumed that would be the reason to do it that way. That's alright, if we can use that to initialize Reconciler.Errors it would be good, as i feel reconcilershould have a function for adding errors upon which requeuing is done.

Reply:
ok, makes sense.
Now that we have the Errors struct we can build this on top then.